--- xerces-c/src/xercesc/util/RefHash3KeysIdPool.c.orig	2003-07-09 14:52:32.000000000 +0200
+++ xerces-c/src/xercesc/util/RefHash3KeysIdPool.c	2003-07-09 14:52:41.000000000 +0200
@@ -118,7 +118,7 @@
     initialize(modulus);
 
     // create default hasher
-    fHash = new (fMemoryManager) HashXMLCh();
+    fHash = new HashXMLCh();
 
     //
     //  Allocate the initial id pointers array. We don't have to zero them
--- xerces-c/src/xercesc/dom/impl/DOMDeepNodeListPool.c.orig	2003-07-09 14:54:10.000000000 +0200
+++ xerces-c/src/xercesc/dom/impl/DOMDeepNodeListPool.c	2003-07-09 14:53:52.000000000 +0200
@@ -90,7 +90,7 @@
     initialize(modulus);
 
     // create default hasher
-    fHash = new (fMemoryManager) HashPtr();
+    fHash = new HashPtr();
 
     //
     //  Allocate the initial id pointers array. We don't have to zero them
