--- xerces-c-3.1.1/m4/xerces_netaccessor_selection.m4.orig	2009-11-12 06:57:31.000000000 +0100
+++ xerces-c-3.1.1/m4/xerces_netaccessor_selection.m4	2010-10-10 10:08:42.231734339 +0200
@@ -26,8 +26,8 @@
 	if test x"$network" = xyes; then
 
 	# Checks for libraries.
-	AC_CHECK_LIB([socket], [socket])
-	AC_CHECK_LIB([nsl], [gethostbyname])
+	AC_SEARCH_LIBS([socket], [socket])
+	AC_SEARCH_LIBS([gethostbyname], [nsl])
 
 	######################################################
 	# Test for availability of each netaccessor on this host.
