--- xerces-c-src_2_7_0/obj/Makefile.in~	2006-02-10 15:29:31.000000000 +0100
+++ xerces-c-src_2_7_0/obj/Makefile.in	2006-02-10 15:57:06.922779750 +0100
@@ -223,7 +223,7 @@
 ifeq (${OS390BATCH},1)
 	@echo "Nothing to do for OS390BATCH ..."
 else
-	-mkdir -p ${DESTDIR}${PREFIX}/lib
+	-mkdir -p ${DESTDIR}${PREFIX}/lib64
 ifeq (${PLATFORM}, CYGWIN)
 	-mkdir -p ${DESTDIR}${PREFIX}/bin
 	$(CP) ${FQ_SO_NAME} ${DESTDIR}${PREFIX}/bin
@@ -234,11 +234,11 @@
 	$(CP) $(XML_LIB_DIR)/${LIBDEPDOM}${SO_TARGET_VERSION}${SHLIBSUFFIX}.a ${DESTDIR}${PREFIX}/lib
 	(cd ${DESTDIR}$(PREFIX)/lib; rm -f ${LINK_DEPDOM}.a libxercesdepdom.dll.a; ln -s ${LIBDEPDOM}${SO_TARGET_VERSION}${SHLIBSUFFIX}.a ${LINK_DEPDOM}.a; rm -f libxercesdepdom.dll.a; ln -s ${LIBDEPDOM}${SO_TARGET_VERSION}${SHLIBSUFFIX}.a libxercesdepdom.dll.a)
 else
-	$(CP) ${FQ_REAL_NAME} ${DESTDIR}$(PREFIX)/lib
-	(cd ${DESTDIR}$(PREFIX)/lib; rm -f ${SO_NAME}; ln -s ${REAL_NAME} ${SO_NAME}; rm -f ${LINK_NAME}; ln -s ${REAL_NAME} ${LINK_NAME})
+	$(CP) ${FQ_REAL_NAME} ${DESTDIR}$(PREFIX)/lib64
+	(cd ${DESTDIR}$(PREFIX)/lib64; rm -f ${SO_NAME}; ln -s ${REAL_NAME} ${SO_NAME}; rm -f ${LINK_NAME}; ln -s ${REAL_NAME} ${LINK_NAME})
 	
-	$(CP) ${FQ_REAL_DEPDOM} ${DESTDIR}$(PREFIX)/lib
-	(cd ${DESTDIR}$(PREFIX)/lib; rm -f ${SO_DEPDOM}; ln -s ${REAL_DEPDOM} ${SO_DEPDOM}; rm -f ${LINK_DEPDOM}; ln -s ${REAL_DEPDOM} ${LINK_DEPDOM})	
+	$(CP) ${FQ_REAL_DEPDOM} ${DESTDIR}$(PREFIX)/lib64
+	(cd ${DESTDIR}$(PREFIX)/lib64; rm -f ${SO_DEPDOM}; ln -s ${REAL_DEPDOM} ${SO_DEPDOM}; rm -f ${LINK_DEPDOM}; ln -s ${REAL_DEPDOM} ${LINK_DEPDOM})	
 endif
 ifeq (${PLATFORM},OS390)
 	$(CP) ${LIBNAME}${VER}${OS390SIDEDECK} ${DESTDIR}$(PREFIX)/lib
