--- xerces-c-src_2_7_0/src/xercesc/runConfigure.orig	2005-09-07 17:55:53.000000000 +0200
+++ xerces-c-src_2_7_0/src/xercesc/runConfigure	2006-02-10 21:49:30.877897750 +0100
@@ -270,7 +270,7 @@
       elif test $platform = "os390"; then
          debugflag="-DNDEBUG";         
       elif test $platform = "linux"; then
-         debugflag="-w -O2 -DNDEBUG";
+         debugflag="-w -DNDEBUG";
       else
          debugflag="-w -O -DNDEBUG";
       fi
